### ISSUE TITLE
Forcible /g00 adware insertion on newspaper websites

### ISSUE THREAD

**Issue by uBlock-user** [uBlockOrigin/uAssets#227]
### URL(s) where the issue occurs
orlandosentinel.com
sandiegouniontribune.com
sun-sentinel.com
mcall.com
boston.com

Those are the ones I have seen so far, there may be more.

### Describe the issue

Forcibly inserts g00 adware content and abuses window.location API if blocked by a filter like `/g00^$important` until it turns into a bad request.

### Screenshot(s)

https://i.gyazo.com/86ab54811f6aaa1785b3d308566d6af6.png

### Versions

- Browser/version: [here] Chromium 57
- uBlock Origin version: [here] 1.10.0

### Settings

Default

### Notes

1) This didn't happen when I visited the website few days ago, however it seems the website is infested with adware as of today, as it keeps trying to load the /g00 stuff when it fails the first time and it also inserts shitty adware cookies too.

2) Blocking inline script does stop the onslaught attack of /g00 however breaks pictures from loading and possibly other things.

**Comment by gorhill** [comment]
The `/g00` stuff is Instart Logic's crap. I have no problem with perceiving this as crapware -- their code goes out of its way to work against end users.

**Comment by gorhill** [comment]
>  is there any solution for this

I will be able to answer when I have the time to investigate.

**Comment by IsraeliAdblocker** [comment]
I've investigated  Instart Logic's crap for the past 3 hours, I now know how they work, how they communicate, how they implement on new customers and etc.
I have a lot of information to reveal and I know the best non "cat & mouse" solution to fight them that we can implement right now.

@gorhill , If you can arrange private channel maybe an invisible thread on issues.adblockplus.org (just give me access I am using the same username there) I will post there all the details.

I don't want Instart Logic people to see my research report.

**Comment by uBlock-user** [comment]
@IsraeliAdblocker Please do. If these guys find success, soon all other major blogs and websites will be infested with it and we will be forced to block inline scripts every now and then, so far folks at easylist forum came up with a filter which no longer works and only worsens the situation.

**Comment by okiehsch** [comment]
`example.com##script:inject(abort-on-property-write.js, I10C)` works on my end.
Example:
Go to `http://www.sandiegouniontribune.com`
You will get peppered with g00 requests.
Now add` sandiegouniontribune.com##script:inject(abort-on-property-write.js, I10C)`
no more g00 requests, at least on my end.
Should work with all the mentioned domains.

**Comment by gorhill** [comment]
> example.com##script:inject(abort-on-property-write.js, I10C) works on my end

Tried first site in list, `orlandosentinel.com`, and the site is rather broken, images won't display.

I suggest:

    orlandosentinel.com##script:inject(wowhead.com.js)

Will await feedback.

**Comment by okiehsch** [comment]
`orlandosentinel.com` displays fine on my end, nothing appears broken,
anyway, if I go to `orlandosentinel.com`, then clear cookies, add
`orlandosentinel.com##script:inject(wowhead.com.js)`,
then reload this is the logger output filtered for "g00"
![g002](https://cloud.githubusercontent.com/assets/16838044/21291710/c5d746c0-c4ea-11e6-9adf-2d871efdfefe.PNG)

and this after I add `orlandosentinel.com##script:inject(abort-on-property-write.js, I10C)`
![g003](https://cloud.githubusercontent.com/assets/16838044/21291717/06f04e04-c4eb-11e6-903d-d2fba90a0e4c.PNG)
and like I said all the pictures display just fine on my end.

**Comment by okiehsch** [comment]
The only mentioned site that doesn´t work on my end is `boston.com`,
but the issue there seems to be the filter `boston.com##script:inject(i10c-defuser.js)`
in `uBlock filters` list, if I disable it, it also works on my end.

**Comment by uBlock-user** [comment]
`sandiegouniontribune.com##script:inject(abort-on-property-write.js, I10C)

That does stop the onslaught attack /g00 attack, however manipulates and adds /g00/refferr/i to the domain at the address bar, and still adds referrer tracking cookies.

**Comment by okiehsch** [comment]
Did you clear the cookies before you added the filter?
Because I don` see any.
![g004](https://cloud.githubusercontent.com/assets/16838044/21291821/1f34d6a8-c4ee-11e6-830f-6d55e01f8fb5.PNG)

Neither is there anything added to the domain.
![g005](https://cloud.githubusercontent.com/assets/16838044/21291831/6c602810-c4ee-11e6-970e-d462f408c107.PNG)

**Comment by uBlock-user** [comment]
Yes I did, I have page opened in another tab. Please let me finish what I'm testing.

**Comment by uBlock-user** [comment]
Tested with `sandiegouniontribune.com##script:inject(i10c-defuser.js)`

same result as wowhead, ads manage to laod after few secs and crap cookies are being inserted.

**Comment by IsraeliAdblocker** [comment]
@gorhill , Please tell me how can I privately share my research with you?

**Comment by okiehsch** [comment]
@gorhill I can, not reliably, sometimes it works, reproduce a broken `orlandosentinel.com` 
 with the filter
 `orlandosentinel.com##script:inject(abort-on-property-write.js, I10C)`.

So my previous post was inaccurate. The reason that I couldn´t reproduce was that I didn´t
realize that I used Chrome 49 on that computer.
I still can never reproduce a broken `orlandosentinel.com` with Chrome 49, but I can with Chrome 55.
Sorry for any confusion I caused.

**Comment by gorhill** [comment]
`orlandosentinel.com##script:inject(wowhead.com.js)` works fine on my side: the first load, the page will redirect to a non-g00 version eventually, as it thinks the console is opened, and as a result the Instart Logic code stops doing crappy things (not unlike cockroaches running for hiding spots when turning on the light):

![a](https://cloud.githubusercontent.com/assets/585534/21320653/cf33cefa-c5df-11e6-90f1-99c5b19c45b6.png)

There are instances of URL with g00 in it, but it's just the URL of the document itself.

**Comment by okiehsch** [comment]
I now tested `orlandosentinel.com##script:inject(wowhead.com.js)`  and
`sandiegouniontribune.com##script:inject(abort-on-property-write.js, I10C)`
with Chrome 49, Chrome 55, Firefox 50.1.0 and Microsoft Edge.
The problem you describe only occurs on Chrome 55 for me.
It works fine on all other browsers.
My OS is Windows 10.

**Comment by gorhill** [comment]
@IsraeliAdblocker See <https://github.com/gorhill/uBlock/issues/1930#issuecomment-268005424>.

**Comment by uBlock-user** [comment]
Tried this `boston.com,mcall.com,sun-sentinel.com,sandiegouniontribune.com,orlandosentinel.com##script:inject(wowhead.com.js)
`
With that filter, I tested, both orlando and sandiego, the first load is very slow and takes a lot of time for the loading spinner to stop and still created some g00 cookies. After refreshing the site upto 3 or 4 times, it becomes normal. /g00 redirecitn is still there, however it's like a popup defuser, it comes when you click and the URL resets back immediately like it never happened, atleast the website is browsable now.

**Comment by gorhill** [comment]
The `boston.com` one still exhibits the issue with the `wowhead.com.js` scriplet (I will rename more appropriately eventually). I am investigating -- I added a scriptlet which defuses Instart Logic's ability to detect that the console is opened, so I can freely investigate using dev tools now.

**Comment by uBlock-user** [comment]
It seems spoofing user agent string to Firefox's ones works perfectly on Chromium. I'm using uMX for spoofing Firefox's UA and it does the job too. So only Chromium based browsers are affected by this.

**Comment by gorhill** [comment]
> So only Chromium based browsers are affected by this.

Yes: <https://np.reddit.com/r/wow/comments/5exq2d/wowheadcom_sucking_bandwidth/dagbmie/>. The server will serve a different document if Firefox (or "not Chrome").

**Comment by gorhill** [comment]
Essentially, the g00 URLs are obfuscated URLs to 3rd-parties that would normally be blocked by blockers:

![a](https://cloud.githubusercontent.com/assets/585534/21322587/d3541398-c5e7-11e6-8c72-5dd0defd6635.png)

**Comment by uBlock-user** [comment]
And those 3rd party urls leave their crap cookies with the help of the script which inserts the cookies as first party ? I already have the Block 3rd party cookies and site data activated, so that's the only way around to insert 3rd party data onto my browser.

**Comment by gorhill** [comment]
> And those 3rd party urls leave their crap cookies with the help of the script which inserts the cookies as first party ?

Looks like this.

**Comment by gorhill** [comment]
I find `||boston.com^$inline-script` seems to work fine.

**Comment by okiehsch** [comment]
There is no obvious site breakage but `www.boston.com/video` will not work, if you disable
inline-script.
There is no "g00" crap in the sourcecode of `www.boston.com/video`, so if you add
`@@||www.boston.com/video$inline-script,domain=www.boston.com`
the video site works.

**Comment by uBlock-user** [comment]
Except for boston, the rest of the lot breaks at root page with thumbnails for the articles and videos at any individual article.

**Comment by IDKwhattoputhere** [comment]
`timeanddate.com` loads g00 too. Noticed when a video ad appeared. `timeanddate.com##script:inject(abort-on-property-write.js, I10C)` seems to work.

**Comment by uBlock-user** [comment]
Another one - `chicagotribune.com`

Blocking the `/g00` profiler did make it browsable, still inserts the same crap as others.

**Comment by gorhill** [comment]
The Instart Logic's code contains a list of sites using their obfuscation scheme, all those listed here are in there -- including chicagotribune.com.

**Comment by gorhill** [comment]
I have a solution, now I have to decide how to make it available. I am thinking of maybe turning uBO-WebSocket into uBO-Extra which would contains all the code which goes beyond filter-based solutions to address some nastiness out there, including the one reported here.

**Comment by uBlock-user** [comment]
I'm not using uBO-WebSocket ext. By turning Websocket into Extra, will users have to install that extension ?

**Comment by gorhill** [comment]
Yes.

**Comment by IsraeliAdblocker** [comment]
Dear Raymond,
I understand your status, but I feeling uncomfortable to reveal my research info to Public eyes.
Please send me an email to: israeliAdblocker@gmail.com, and I will send you what I have, I want you to have that info, you can decide to use it or not at your own choice.

Thanks.

**Comment by gorhill** [comment]
> I feeling uncomfortable to reveal my research info to Public eyes.

Given what Instart Logic's technology does, I think there is a lot of value to make public all your findings. Their technology is extremely hostile to users, as it's also a way to bypass a user's wish to block third-party cookies, or even a user's wish to block undesirable servers using a hosts file. I can see broad public disapproval to the technology and we should not underestimate the shame factor.

**Comment by gorhill** [comment]
uBO-WebSocket has been [renamed uBO-Extra](https://github.com/gorhill/uBO-Extra), with a broader purpose of better meeting user expectations when they use uBO Origin. It takes care of the issue here. Updated in [Chrome store](https://chrome.google.com/webstore/detail/ublock-origin-extra/pgdnlhfefecpicbbihgmbmffkjpaplco) as well.

**Comment by okiehsch** [comment]
Works great with the mentioned sites, however with sites like 
`http://ottawacitizen.com` or `http://www.thomson.co.uk`
it breaks part of the functionality of the sites.
In the case of `http://ottawacitizen.com` you can't use the search function or sign in.
![un](https://cloud.githubusercontent.com/assets/16838044/21373952/17ec8b48-c723-11e6-8021-383e6404be8a.PNG)

In the case of `http://www.thomson.co.uk` you can't use the interactive boxes.
![unbent](https://cloud.githubusercontent.com/assets/16838044/21374066/14b6f3cc-c724-11e6-8bfe-3070413a962a.PNG)
Other sites with that issue
`montrealgazette.com,		
calgaryherald.com,	 
edmontonjournal.com,	
theprovince.com,	
windsorstar.com,
firstchoice.co.uk, 
leaderpost.com,
thestarphoenix.com,	
falconholidays.ie`

**Comment by gorhill** [comment]
Thanks @okiehsch, I will investigate the issues.

- `ottawacitizen.com`: I did see the `HtmlStreaming` issue.
- `www.thomson.co.uk`: looked fine (now it's "undergoing essential maintenance"). What was broken exactly?
- `montrealgazette.com`: did not see the `HtmlStreaming` issue.
- `calgaryherald.com`: did not see the `HtmlStreaming` issue.
- `edmontonjournal.com`: did not see the `HtmlStreaming` issue.
- `theprovince.com`: did not see the `HtmlStreaming` issue.
- `windsorstar.com`: did not see the `HtmlStreaming` issue.
- `firstchoice.co.uk`: Site is "undergoing essential maintenance".
- `leaderpost.com`: did not see the `HtmlStreaming` issue.
- `thestarphoenix.com`: did not see the `HtmlStreaming` issue.
- `falconholidays.ie`: Site is "undergoing essential maintenance".

**Comment by okiehsch** [comment]
- `www.thomson.co.uk` see second screenshot of my previous post.

- `montrealgazette.com` same issue as `ottawacitizen.com`, can still reproduce
![unbenann](https://cloud.githubusercontent.com/assets/16838044/21375190/37d36662-c72c-11e6-9b57-ab87289bc3af.PNG)
Same is true for the rest, except for the sites undergoing maintenance.

For example the console for `leaderpost.com`
![unbenannt1](https://cloud.githubusercontent.com/assets/16838044/21375353/34781a0c-c72d-11e6-910e-e1bd16f8e10e.PNG)

**Comment by gorhill** [comment]
List of sites gathered from the IL's g00 script:

    about.com
    applyabroad.org
    boston.com
    cargurus.com
    chroniclelive.co.uk
    cnet.com
    corriere.it
    gamepedia.com
    mmo-champion.com
    twincities.com
    edmunds.com
    foxnews.com
    gamerevolution.com
    holidaycheck.de
    i10c.net
    infinitiev.com
    instarttest.com
    drudgereport.com
    headlinepolitics.com
    refdesk.com
    tellmenow.com
    thepoliticalinsider.com
    tmn.today
    legacy.com
    metal-hammer.de
    msn.com
    nasdaq.com
    photobucket.com
    calgaryherald.com
    calgarysun.com
    canoe.com
    edmontonjournal.com
    edmontonsun.com
    financialpost.com
    ifpress.com
    leaderpost.com
    montrealgazette.com
    nationalpost.com
    ottawacitizen.com
    ottawasun.com
    theprovince.com
    thestarphoenix.com
    torontosun.com
    vancouversun.com
    windsorstar.com
    winnipegsun.com
    ranker.com
    reshadi.com
    saveur.com
    sherdog.com
    slickdeals.net
    space.com
    buzznet.com
    celebuzz.com
    deathandtaxesmag.com
    gofugyourself.com
    idolator.com
    spin.com
    stereogum.com
    thefrisky.com
    thesuperficial.com
    vibe.com
    sporcle.com
    sportingnews.com
    testdomain.com
    thinkfu.com
    timeanddate.com
    tronc.com
    baltimoresun.com
    capitalgazette.com
    carrollcountytimes.com
    chicagotribune.com
    citypaper.com
    courant.com
    ctnow.com
    dailypress.com
    delmartimes.com
    discoversd.com
    growthspotter.com
    hoylosangeles.com
    lajollalight.com
    latimes.com
    mcall.com
    orlandosentinel.com
    ranchosantafereview.com
    redeyechicago.com
    sandiegouniontribune.com
    southflorida.com
    sun-sentinel.com
    vagazette.com
    trustedreviews.com
    washingtonpost.com
    weather.com
    destinydb.com
    hearthhead.com
    lolking.net
    mmoui.com
    opshead.com
    wowhead.com
    zam.com
    computershopper.com
    extremetech.com
    geek.com
    ign.com
    logicbuy.com
    pcmag.com
    speedtest.net

**Comment by gorhill** [comment]
> `montrealgazette.com` same issue as `ottawacitizen.com`, can still reproduce

Looks like I might be served a different document, there is no instance of `HtmlStreaming` on my side in the source code, no such error at the console.

**Comment by okiehsch** [comment]
Seems to be the case
![unbenannt2](https://cloud.githubusercontent.com/assets/16838044/21375713/af3f8dea-c72f-11e6-9e4a-7a7351a0f1d7.PNG)

**Comment by gorhill** [comment]
Ok I understand, they are browser-sniffing, and the g00 javascript is not served with Chrome 57, but occurs with Chromium 53 (I use Chrome to test uBO with default settings).

**Comment by okiehsch** [comment]
After a rather crude search here:
`https://publicwww.com/websites/i10c.morph/`
All the sites on that list have at least remnants of " Instart Logic" in their source code.

**Comment by uBlock-user** [comment]
uBO-Extra works as intended, thanks a lot. Still doublclick.net and google-analytics's channel ID cookie was placed, can you do something about that ?

**Comment by gorhill** [comment]
I modified the approach re. g00: the g00-busting code will be injected only on sites for which it has been tested as working. Currently there is two version of the g00-busting scriptlet: one is the same as published yesterday, the other one is specific to those sites above using `HtmlStreaming`. The g00-busting scriplet will be injected **only** on site for which it is tested and confirmed as working as intended. Thus, report here any site which must be added.

**Comment by IDKwhattoputhere** [comment]
uBO-Extra 2.0 works on `timeanddate.com`, uBO-Extra 2.1 does not. I assume it should be added to the list of working sites then.
Is there an easier way to test sites than switching between 2.0 and 2.1?

**Comment by gorhill** [comment]
> Is there an easier way to test sites than switching between 2.0 and 2.1?

Best is to load the extension locally, and add sites to the list in the code, then restart the extension to see if it works. There are two scriptlet versions for g00 stuff, uBO-Extra v2.1 contains both, while v2.0 contains only one. If `view-source:` for the site shows that it used `HtmlStreaming`, the second scriptlet is probably the one to use. Since `timeanddate.com` worked with v2.0, than the first version is the one to use.

In retrospect it was bad to apply the fix indiscriminately.

**Comment by IsraeliAdblocker** [comment]
@gorhill have you approached me via email? I've sent a response but now I have second thoughts that this weren't you, just checking.

**Comment by gorhill** [comment]
> @gorhill have you approached me via email?

I did not contact you by email. For personal reasons, I abstain from using my personal email for anything related to GitHub (except for _very_ rare exceptions), this is also why I pretty much never answer emails sent to me as a result of my presence on GitHub.

**Comment by IsraeliAdblocker** [comment]
OK, so I was contacted by Instart Logic with this email:
`What do you want to share with me about the g00 stuff?
Again keep in mind the HTML/CSS/JS code is open to anybody.
 
--
gorhill`

from this address:
raymond.hill@usa.com

Now they have the info :) and we are sure to know that they are reading this thread and chaining everything by our comments.

**Comment by IsraeliAdblocker** [comment]
This is the mail I've sent, now that I know that this is probably them that opened this email:

```I am not talking about the HTML/CSS/JS code that is opened to everyone.
I am aware of the fact that there are strings of domains that they are working with in their code.
But I've found out, that there is a single URL which you can test and find if any website has been using Instart logic tech, even those who are testing their tech in their test environment.

This URL is a good example:
http://d3btzwrpys5idxnpbmvzcy5maw5hbmnpywxwb3n0lmnvbq00.g00.zam.com/g00/2_d3d3LmZpbmFuY2lhbHBvc3QuY29t_/TU9SRVBIRVVTMiRodHRwOi8vd3BtZWRpYS5idXNpbmVzcy5maW5hbmNpYWxwb3N0LmNvbS8yMDE0LzA1L3R3ZWVkLmpwZz93PTMwMCZpMTBjLm1hcmsuaW1hZ2UudHlwZQ%3D%3D_$/$/$/$

This is an example for zam.com, all that you need to do in order to check different website is to change this to something else.
I've checked this URL against the top 10K alexa domains plus the strings from their JS file and those are the domains that they are working with correctly. 
ottawacitizen.com
montrealgazette.com
leaderpost.com
thestarphoenix.com
theprovince.com
vancouversun.com
calgaryherald.com
edmontonjournal.com
windsorstar.com
orlandosentinel.com
sandiegouniontribune.com
sun-sentinel.com
mcall.com
Boston.com
Edmunds.com
pcmag.com
LolKing.net
Wowhead.com
torrentsgroup.com
nasdaq.com
saveur.com
financialpost.com
montrealgazette.com
ottawacitizen.com
ottawasun.com
vancouversun.com
windsorstar.com
celebuzz.com
deathandtaxesmag.com
timeanddate.com
capitalgazette.com
financialpost.com
weather.com
ign.com
logicbuy.com
geek.com
extremetech.com
computershopper.com
zam.com
opshead.com
hearthhead.com
destinydb.com
redeyechicago.com
latimes.com
dailypress.com
courant.com
citypaper.com
carrollcountytimes.com
baltimoresun.com
sportingnews.com
sporcle.com
vibe.com
thesuperficial.com
thefrisky.com
stereogum.com
spin.com
gofugyourself.com
celebuzz.com
sherdog.com
ranker.com
winnipegsun.com
torontosun.com
thestarphoenix.com
nationalpost.com
edmontonsun.com
canoe.com
calgarysun.com
photobucket.com
legacy.com
msn.com
tmn.today
infinitiev.com
holidaycheck.de
gamerevolution.com
edmunds.com
twincities.com
mmo-champion.com
gamepedia.com
cnet.com
chroniclelive.co.uk
about.com


as you can see, the list contains 5 of the top 100 websites in the world:
msn.com
about.com
cnet.com
ign.com
weather.com


Moreover, I have a rule that will win the game, a rule that will work in both Ublock and ABP.

if we will do this:

@@||sporcle.com^$generichide
@@|http://$image,domain=sporcle.com,third-party
||*.sporcle.com^$subdocument

we will kill the way that they detecting Adblock while blocking their iframes.
in that way, if they will change the way that they detect Adblock, they will be the ones who will destroy all of their websites functionality, so the websites owners won't allow it and Instart Logic will be stuck.
so they will have no option.

If they will change the way that they detect Adblockers and their Ads will appear, the $subdocument rule will force their code to destroy the site (with their hands, since they are doing this while listening to the iframe onerror).

If they won't change the way that they detect Adblockers, they will have to remove the part that is destroying the site before changing their inline script, and then our $subdocument rule will do it's part again.```

**Comment by uBlock-user** [comment]
You should be more careful before answering e-mails. He doesn't have any e-mail address at usa.com, it's sent from an e-mail spoofing server. Check the message source and you will find out the origin of the server from where the e-mail arrived, :D 

Whoever it is, judging from the e-mail, is keeping track of things on this issue tracker.

**Comment by okiehsch** [comment]
The following sites will all display
`Uncaught TypeError: Cannot read property 'HtmlStreaming' of undefined`
in the console and therefore potentially break some functionality of the site,
if you use uBO-Extra v2.0.
`edmunds.com;
katespade.com;
nationalpost.com;
homes.com;
hayneedle.com;
thrivemarket.com;
lastcall.com;
bergdorfgoodman.com;
journeys.com;
rockler.com;
thomson.co.uk;
montrealgazette.com;
ottawacitizen.com;
calgaryherald.com;
livingspaces.com;
barenecessities.com;
stelladot.com;
thereformation.com;
edmontonjournal.com;
theprovince.com;
viking-direct.co.uk;
canada.com;
driving.ca;
windsorstar.com;
horchow.com;
officedepot.fr;
smartpakequine.com;
firstchoice.co.uk;
esalerugs.com;
petflow.com;
visiondirect.com;
pens.com;
woodworkersjournal.com;
abchome.com;
leaderpost.com;
essie.com;
penseurope.com;
thestarphoenix.com;
viking.es;
vikingop.it;
vikingdirect.ie;
vikingdirect.nl;
falconholidays.ie;
irugs.co.uk;
hockeyinsideout.com;
opshead.com;
leonardo.com;
rugstudio.com;
vikingdirect.fr;
ghurka.com;
pens.jp;
vikingdirect.be`

Many of these sites do not serve any "g00" crap, but all of them have 
`I10C.HtmlStreaming.PatchInit` in their source codes which will lead to the aforementioned
error in the console with uBO-Extra v2.0.

`thestarphoenix.com`, `edmontonjournal.com` and `montrealgazette.com` are special cases, atleast on my end,
because they will not display at all with uBO-Extra v2.0 or uBO-Extra v2.1,
even i I whitelist them.

With  uBO-Extra v2.0 I get the expected `Uncaught TypeError: Cannot read property 'HtmlStreaming'`
With uBO-Extra v2.1 I get
![unbena](https://cloud.githubusercontent.com/assets/16838044/21402509/1f8b123e-c7b9-11e6-8966-a813dc59d380.PNG)


The other sites mentoned in `Instart Logic buster: v2` work fine.

**Comment by gorhill** [comment]
> With uBO-Extra v2.1 I get

The error seems to come from a `pubads_...` script (which I believe is from Google Ads). I do not see such script loaded on my side. What does the logger says? Are there any exception filters on your side as a result of using non-default filter lists?

**Comment by okiehsch** [comment]
As I said this happens even if I whitelist the side as indicated by the grey uBlock symbol in the screenshot.
The logger predictably shows nothing.
![montre](https://cloud.githubusercontent.com/assets/16838044/21405601/38320f28-c7c7-11e6-8252-d2bac1662b0b.PNG)

**Comment by ghajini** [comment]
how disconnect and ghostery are preventing g00 stuff from loading?????
i don't see cookie stored in chrome of g00
seems only with ublock origin/adguard/abp

**Comment by IDKwhattoputhere** [comment]
They probably don't prevent it from running but they don't block the requests that g00 uses to detect adblockers. I guess their filter lists aren't as extensive and g00 simply doesn't target them (yet).

**Comment by gorhill** [comment]
> They probably don't prevent it from running but they don't block the requests that g00 uses to detect adblockers.

Yeah, notably Ghostery does not block network requests to ad servers, Ghostery left, uBO right:

![a](https://cloud.githubusercontent.com/assets/585534/21406526/203c0b3a-c79a-11e6-9075-af6dedd42250.png)

**Comment by ghajini** [comment]
also Firefox with ublock origin enabled handles g00 stuff with just 1 ublock privacy filter....
Chrome going crazy

**Comment by gorhill** [comment]
Firefox does not handle g00 stuff, the server sends a different document to Firefox users. Just spoofing the UA to Firefox when using Chrome and no g00 stuff.

**Comment by ghajini** [comment]
Thanx. ..🙋 for Firefox  ublock origin version😊

**Comment by uBlock-user** [comment]
Spoofing UA doesn't stop the onslaught of 3rd party cookies that are fired to the browser though.

**Comment by IsraeliAdblocker** [comment]
so it's an easy solution, just spoof the UA of the user when UBo detect one of those sites.

**Comment by okiehsch** [comment]
@gorhill 
> Best is to load the extension locally, and add sites to the list in the code, then restart the extension to see if it works

This list works on my end.
Instart Logic buster: v1

            'baltimoresun.com',
            'boston.com',
            'capitalgazette.com',	 	 	 
            'carrollcountytimes.com',	 
            'celebuzz.com',
            'chicagotribune.com',
            'courant.com',	 	 	 	 	
            'dailypress.com',	 	 	 	 	
            'deathandtaxesmag.com',	 	 	 	 	
            'gamerevolution.com',	 	 	 	 	
            'gofugyourself.com',	 	 	 	 	
            'hearthhead.com',	 	 	 	 
            'infinitiev.com',
            'mcall.com',
            'nasdaq.com',
            'orlandosentinel.com',
            'ranker.com',
            'sandiegouniontribune.com',
            'saveur.com',
            'sherdog.com',
            'spin.com',
            'sporcle.com',
            'stereogum.com',
            'sun-sentinel.com',
            'thefrisky.com',	 	 	 	 	
            'thesuperficial.com',	 	 	 	 	
            'timeanddate.com',	 	 	 	 	
            'tmn.today',
            'vancouversun.com',
            'vibe.com',
            'weather.com',
            'wowhead.com',
        
Instart Logic buster: v2

            'calgaryherald.com',
            'edmontonjournal.com',
            'edmunds.com',
            'financialpost.com',
            'leaderpost.com',
            'montrealgazette.com',
            'nationalpost.com',
            'ottawacitizen.com',
            'theprovince.com',
            'thestarphoenix.com',
            'windsorstar.com',

**Comment by gorhill** [comment]
@okiehsch Can you submit a pull request for these additions?

**Comment by harshanvn** [comment]
> List of sites gathered from the IL's g00-related script (does not necessarily mean these sites are g00 infested, this will need confirmation):
> 
> extremetech.com

I think this site is using g00 crapware. I see below re-directions.. pulled from the logger. Along, with turning all 3rd party requests/cookies for 1P request/cookies.

> http://www.extremetech.com/
> http://www.extremetech.com/g00/?i10c.referrer=

uBO v1.10.4 + uBO Extra v2.6 on Chrome v55

**Comment by gorhill** [comment]
Fixed with <https://github.com/gorhill/uBO-Extra/releases/tag/2.7>.

**Comment by okiehsch** [comment]
@gorhill , you can also add
`celebslam.com.
computershopper.com.
geek.com.
lolking.net.
mmo-champion.com.
pcmag.com.
twincities.com`
All to Instart Logic defuser v1.

**Comment by ghajini** [comment]
only `celebslam.com`

**Comment by okiehsch** [comment]
They do not serve it to everybody at the same time,
I can still reproduce.
Go to
`view-source:http://www.computershopper.com/g00/?i10c.referrer=`
`view-source:http://geek.com/g00/?i10c.referrer=`
`view-source:http://lolking.net/g00/?i10c.referrer=`
`view-source:http://www.pcmag.com/g00/?i10c.referrer=`
Do you see any "g00" in the source code?

At `mmo-champion.com` they reported the ads in the forum.
`http://www.mmo-champion.com/threads/631902-Advertising-Reporting-bad-ads/page45`

**Comment by uBlock-user** [comment]
Have you uploaded `v2.7 `to chrome store ?

**Comment by jawz101** [comment]
Why not recommend disabling 3rd party cookies in the browser settings?  I've been doing that for years.

**Comment by uBlock-user** [comment]
> Why not recommend disabling 3rd party cookies

I have always disabled 3rd party cookies and site data from the beginning and even after that they bypassed it by sending third-party cookies as first party cookies through an inline-script. It's not as simple as you think.

**Comment by uBlock-user** [comment]
`chicagotribune.com` no longer works, ads are shown and `g00` cookies are being inserted again.

**Comment by gorhill** [comment]
Unable to reproduce. I also tried with a US VPN in case.

**Comment by mapx-** [comment]
fixed in easylist
https://github.com/easylist/easylist/commit/9261f6a493343000916026d1e73aeae440259041

**Comment by gorhill** [comment]
Personally I believe such solution based on such broad exception filter should come with disclosure to users:

> side-effect to the solution: to unblock 3rd-party javascript resources from anywhere which URL has the pattern `.js?&` in it, _including_ from sites listed in your privacy or malware lists you may have enabled.

**Comment by uBlock-user** [comment]
> fixed in easylist

no change, g00 cookies are still being inserted and ads ge loaded. Whitelisting is not going to foil or anything 👎 

https://i.gyazo.com/a345f1e6e3d037020573ccb2811ab620.png

> Unable to reproduce.

I can reproduce it on Chrome 57
https://i.gyazo.com/6f5d7689f4dd5b40eb6b4aa247da342f.png

g00 cookies - https://i.gyazo.com/bf0038ffcae33a3621e7ca9ad4a12831.png

**Comment by gorhill** [comment]
I can reproduce with Chrome 57 (I was trying with Chromium 53).

**Comment by uBlock-user** [comment]
I can reproduce the same on Chromium 55 too and in both cases, I have uBO Extra enabled and uBO version is 1.10.4 the latest WebStore build.

**Comment by gorhill** [comment]
> I can reproduce the same on Chromium 55 too and in both cases, I have uBO Extra enabled

Hopefully someone else will test and report results. I also tested with uBO 1.10.4 and all was fine -- except for the now more numerous exception filters being hit as per logger.

**Comment by uBlock-user** [comment]
> more numerous exception filters being hit as per logger.

Shockingly, this is what I see in the popup panel - https://i.gyazo.com/b315fe066c0f741be286b662bff0a423.png

It looks like as if the entire website has been whitelisted by Easylist

**Comment by gorhill** [comment]
> Shockingly, this is what I see in the popup panel

Weird. This is what I get:

![a](https://cloud.githubusercontent.com/assets/585534/22183260/c1a532f2-e087-11e6-8b73-4c333d625aef.png)

Though for example it appears now `tpc.googlesyndication.com`, `ads.twitter.com` are whitelisted because of broad exception filters such as `@@.ico^$domain=chicagotribune.com`

**Comment by uBlock-user** [comment]
> Weird. This is what I get

Yes, I see that in the popup panel, `only` if I `don't` remove `g00` cookies and `first-party` cookies.

**Comment by DanIbba** [comment]
why you're not using this regex - /.*[/\.]g00[/\.].*/ to block them?

**Comment by uBlock-user** [comment]
> why you're not using this regex 

That's blocks the entire website. If I wanted to block an entire website, I would have done it already.

**Comment by gorhill** [comment]
If one does not mind potential breakage of some site features, blocking inline script is a better solution than any horribly inefficient filters such as  `/.*[/\.]g00[/\.].*/`.

**Comment by DanIbba** [comment]
did u check it for boston.com, it works perfectly.
Its not block the entire website.

**Comment by uBlock-user** [comment]
> did u check it for boston.com, it works perfectly.
Its not block the entire website.

So tell me why do I see this - https://i.gyazo.com/95abc501ee890a091e5f079ef4b2b281.png ?

**Comment by gorhill** [comment]
> So tell me why do I see

Because it's not a good filter, no filter list maintainers would come up with such a bad filter. The same thing can be accomplished with saner filters:

    .g00.
    .g00/
    /g00.
    /g00/*

Or at least the regex could have been made _less_ worst:

    /[/.]g00[/.]/

With that said, I didn't check if above actually work, as said I can't reproduce with uBO-Extra enabled.

**Comment by IDKwhattoputhere** [comment]
> Hopefully someone else will test and report results.

In case there's still interest: Chrome 57 with uBO 1.10.4 + uBlock Origin Extra 2.8 [enabled](https://cloud.githubusercontent.com/assets/16030872/22183713/42c5c66c-e0c4-11e6-8c29-f9954306bca7.png)/[disabled](https://cloud.githubusercontent.com/assets/16030872/22183714/42c7af18-e0c4-11e6-9391-bb0a7e2e27d6.png)

**Comment by okiehsch** [comment]
Same as @IDKwhattoputhere with Chrome 55 on my end.
A websocket connection to literally `example.com` appears to be present  in all sites listed
in  `uBlock Origin Extra 2.8`

**Comment by okiehsch** [comment]
> I can reproduce it on Chrome 57
https://i.gyazo.com/6f5d7689f4dd5b40eb6b4aa247da342f.png

I can reproduce this behaviour with Chrome 55 and `uBlock Origin Extra 2.8` only if I
add the filter `*$websocket,important`  just like @uBlock-user.

**Comment by okiehsch** [comment]
If I add `*$websocket,important` I can reproduce your issue, without it I can't.

**Comment by uBlock-user** [comment]
If I remove the `||$websocket` filter entirely, it works normally and the g00 stuff doesn't happen. Very strange. Good find @okiehsch `||$websocket` is triggering the `g00`.

**Comment by gorhill** [comment]
Looks like they changed their inline script, and their behavior according to whether an attempt at `websocket`-connecting to `example.com` fails or not.

**Comment by uBlock-user** [comment]
They're using `example.com` as a honeypot trap basically

**Comment by okiehsch** [comment]
> Looks like they changed their inline script, and their behavior according to whether an attempt at websocket-connecting to example.com fails or not

There seems to be a difference between simply blocking a websocket-connection and 
blocking websockets via Content Security Policy.
The websocket-connection to `example.com` is blocked by Easylist.
![sockett](https://cloud.githubusercontent.com/assets/16838044/22187416/34b6177e-e106-11e6-9f57-8f22b1121d63.PNG)
![sockt](https://cloud.githubusercontent.com/assets/16838044/22187520/58cbbc20-e108-11e6-9da6-eb07984abfb5.PNG)
This does not result in a redirection and the following "g00" crap.

If I use the CSP I get redirected and bombarded with "g00" requests and cookies.
![soc](https://cloud.githubusercontent.com/assets/16838044/22187529/7f32f55e-e108-11e6-8597-38b121dad346.PNG)

**Comment by gorhill** [comment]
Blocking through CSP causes an exception, as opposed to just blocking using the wrapper.

**Comment by gorhill** [comment]
> Add `||$websocket` or `||$websocket,important` to reproduce.

@uBlock-user Could you look if the issue is fixed with 2.9?

**Comment by okiehsch** [comment]
@gorhill 2.9 does not work with a default setup. I get redirected on all websites I tried.
![insta](https://cloud.githubusercontent.com/assets/16838044/22400304/4e15c7dc-e5b1-11e6-8744-d95df3d812d5.PNG)

It works fine with 2.8

**Comment by gorhill** [comment]
> I get redirected on all websites I tried.

There is a first redirection, but the site eventually configure itself to no longer use the g00 stuff since it thinks the console is opened. The websocket  @ example.com was for the IL to detect whether uBO-Extra was installed -- this is fixed now.

**Comment by okiehsch** [comment]
I agree, but looking at the logger is rather confusing now, because it **looks** like you are redirected.

My logger after randomly navigating through chicagotribune.com.
![instnnt](https://cloud.githubusercontent.com/assets/16838044/22400497/e5f87f90-e5b6-11e6-9201-36d22c77dea3.PNG)

**Comment by gorhill** [comment]
Ok, this is odd.

If I understand the IL code was trying to see if the WebSocket wrapper was installed, and if so it apparently was disabling itself _completely_. Now the change make it so that IL's websocket detection code is foiled -- hence it no longer disables itself completely. The CSP directive to foil websocket was also causing the detection code to not work.

The trapping of the `I10C` property is not what actually work, what works is merely to make sure the IL websocket wrapper detection code is triggered. Not sure what they are trying to achieve. I will modify to ensure the detection code work in all case, including when using a CSP.

**Comment by uBlock-user** [comment]
@gorhill Sorry for the delay. Have you uploaded v2.9 to chrome store ? I'm still on v2.8.

**Comment by uBlock-user** [comment]
@gorhill Tested with v2.9 by loading it manually and it works fine with `||$websocket` filter, back to normal behavior 👍

**Comment by okiehsch** [comment]
@gorhill `calgarysun.com` and `torontosun.com` use Instart Logic code and you would expect to get "g00"cookies ads etc.
At least on my end, this is not the case with `uBO-Extra` enabled, even though `calgarysun.com` and
`torontosun.com` are not part of `contentscript.js`.
If you disable `uBO-Extra` you get the expected redirection.
This happens with every `uBO-Extra` version up to 2.2 with version 2.1 I get the 
typical redirection and subsequent cookies and ads.
If you delete the websites mentioned in `contentscript.js` they will still be "g00" free if you have 
`uBO-Extra` enabled.
The downside to this, at least on my end, is that all sites mentioned under `Instart Logic buster: v2`
have some broken functionality, for example the search function is broken on `calgaryherald.com`
unless I disable `uBO-Extra`. 
Console for `calgaryherald.com`
![calt](https://cloud.githubusercontent.com/assets/16838044/23104670/b056ccfc-f6d2-11e6-9fd2-efcee446649c.PNG)

**Comment by gorhill** [comment]
To be sure I understand correctly, with uBO-Extra v2.11:

1) Removing all sites for Instart Logic buster v1 and all is still fine for all targeted sites?

2) Instart Logic buster v2 causes problem on your side (the errors in dev console) on enumerated sites? I tried with `calgaryherald.com`, `http://ottawacitizen.com/` and I can't see these errors.

I noticed that a lot of broad exception filters have been added to EasyList for these sites.

**Comment by okiehsch** [comment]
1. I removed a few sites for example `twincities.com` and they are still "g00" free on my end.
2. All Instart Logic buster v2 sites have the same errors in the console on my end.
3. Do you get redirected and the subsequent "g00" cokkies etc if you go to `torontosun.com`
with uBO-Extra enabled?

**Comment by ghajini** [comment]
`ottawacitizen.com` is blank on loading...
`http://prnt.sc/eas349`

dev console=

`http://prnt.sc/eas4hf`
`http://prnt.sc/eas4uj`
http://prnt.sc/eas58i

**Comment by ghajini** [comment]
@okiehsch 
1.search and video panel broken
`http://ottawacitizen.com/news/local-news/the-questions-and-answers-what-you-need-to-know-about-the-opioid-crisis`

2. basic layout,site is broken
`http://ottawacitizen.com/news/local-news/one-bad-pill-parents-of-ottawa-teen-killed-by-opioid-overdose-speak-out`
![screenshot 148](https://cloud.githubusercontent.com/assets/20338483/23333366/61d2712a-fbb0-11e6-8f5f-0f2af8bcd218.png)

3.site is blank on loading
`http://calgaryherald.com/news/crime/social-worker-wants-alex-alerts-to-protect-at-risk-children-from-vanishing`
![screenshot 149](https://cloud.githubusercontent.com/assets/20338483/23333386/a3e61cc4-fbb0-11e6-8475-f649dcdcea50.png)

4.
videos broken on nationalpost.com and all subsites like
here ''news videos''
`http://business.financialpost.com/news/retail-marketing/how-costco-canada-breaks-retail-rules-to-win`

**Comment by uBlock-user** [comment]
@ghajini That's just you, everything works fine here, search functions as expected, videos load and play. I cannot reproduce any of that.

**Comment by uBlock-user** [comment]
@ghajini You didn't mention Chrome version. Only Chromium 56 is affected.

**Comment by ghajini** [comment]
yes i use only stable chrome version
affected browser= google chrome Version 56.0.2924.87 (64-bit)
ublock 1.11
ubo extra 2.11

**Comment by uBlock-user** [comment]
annnnnnd the g00 shit is back again, uBO Extra v 2.12 no longer working again on `sandiegouniontribune.com` and `spin.com`. I doubt I need to check others as these two are already sprawling back again.

**Comment by ghajini** [comment]
@mapx- & @smed79
any idea why easylist allowing/whitelisting ad networks on these websites???

**Comment by SMed79** [comment]
@ghajini In most of the cases to fix an anti adblock or kill these ads pushed if an adblock is detected.

**Comment by ghajini** [comment]
@gorhill 
on main homepage video not playing..nothing happen when clicking under video...
ublock origin,uBO extra all updated to latest
same problem with
`calgaryherald.com`
`http://www.financialpost.com/index.html`
`http://www.nationalpost.com/index.html`
and possibly other domains also...

`sometime blank loading of these websites`

**Comment by X01X012013** [comment]
Hey guys, after some testing, it looks like it only affect Chrome, the website works normally in Opera. 
Looks like locking `navigator` to an empty object stops the exploit, but also stops the comment section. When Instart Logic said it's a breakthrough, they are not joking... This script will be so easy to kill if we have `beforescriptexecute` event...

**Comment by gorhill** [comment]
> on main homepage video not playing..nothing happen when clicking under video...

Unable to reproduce with uBO-Extra 2.12 and latest uBO + default settings.

**Comment by gorhill** [comment]
> This exploit will be so easy to kill if we have beforescriptexecute event... But their target, Chrome, doesn't have this event

It's being also deprecated in Firefox, it will no longer be available with WebExtensions.

However, I experimented with using AAK's trick to mimic `beforescriptexecute` in uBO-Extra, and it does seem to work -- currently removing any inline scripts with the string `i10c` in it.

**Comment by ghajini** [comment]
load `www.ottawacitizen.com`
click on video on right side corner
nothing happens
@gorhill 
`http://prnt.sc/ekbj9x`
`http://prnt.sc/ekbkuj`
`http://prnt.sc/ekbl37`
i think ublock extra breaks brightcove on sites due to which video doesn't play....

**Comment by okiehsch** [comment]
@gorhill i I can reproduce with Chrome 57 and uBO Extra 2.12 
If I go to `edmontonjournal.com` or `ottawacitizen.com` "TODAY'S HEADLINE VIDEOS" is visible,
but if I click to play it "disappears" and this error appears in the console log.

![edm](https://cloud.githubusercontent.com/assets/16838044/23962321/0457c76e-09ae-11e7-95ac-c9b1b3863906.JPG)

This occurs even if I whitelist the site. If I disable uBO Extra the video plays fine.

**Comment by X01X012013** [comment]
@gorhill 
That hack kind of work, but not really for pages loaded though `POST`. And it has race conditions, and seems to interfere with uBO element picker. That is usually my last resort. 

BTW, my implementation is: 
```JavaScript
a.patchHTML = function (patcher) {
    a.win.stop();
    GM_xmlhttpRequest({
        method: "GET",
        url: a.doc.location.href,
        synchronous: true,
        headers: {
            "Referer": a.doc.referrer
        },
        onload: function (result) {
            a.doc.write(patcher(result.responseText));
        }
    });
};
```

**Comment by gorhill** [comment]
> it has race conditions, and seems to interfere with uBO element picker.

Not in my implementation. The race condition is probably a result of roundtrip to/from the main extension process. There is a way uBO can avoid this roundtrip for when inline script tag filtering is required.

**Comment by gorhill** [comment]
> if I click to play it "disappears" and this error appears in the console log

Alright, strike everything I said above, I was testing on Chrome 58 and apparently the IL's crap bails out on that browser, as it does on Firefox. I can reproduce the issues with Chromium 56.

**Comment by X01X012013** [comment]
@gorhill That's awesome! If we can filter specific in-line scripts from Chromium based browsers, pretty much all our problems are solved 😄

**Comment by ghajini** [comment]
Thanx @gorhill....you understand finally....what about `revolving/rotating ad networks` above sites are using......easylist is just throwing broad exceptions...what we can do for this???

**Comment by uBlock-user** [comment]
v2.15b0 appears to be working, can they bypass this too though ?

**Comment by TheEnhas** [comment]
http://www.idigitaltimes.com/breath-wild-armor-sets-best-and-most-practical-outfits-game-587321

Using Google Chrome 58 (beta), uBO 1.11.5b2 and uBO-Extra 2.15b0, the site seems to circumvent it going by the endless /g00 spam in the logger.

**Comment by uBlock-user** [comment]
@gorhill It's happening again, they patched your latest patch in v2.15, uBO-Extra no longer working.

**Comment by gorhill** [comment]
I need to know what URLs you are talking about.

**Comment by uBlock-user** [comment]
same as always - `sandiegouniontribune.com`

**Comment by gorhill** [comment]
2.15b0 was a special build in a different branch. The changes from it are not in 2.15. I think it's time to bring the changes out of experimental status at this point.

**Comment by uBlock-user** [comment]
Can you reproduce it with 2.15b0 ?

**Comment by gorhill** [comment]
I brought the changes from 2.15b0 in main branch, this fixes the issue.

However, so many exception filters in EasyList, this is disheartening.

**Comment by uBlock-user** [comment]
Ah yes, 2.15b0 works fine. We can just add `$important` for those exceptions for the time-being and when 1.15.0 is released replace them with `$badfilter`.

**Comment by gorhill** [comment]
> We can just add `$important` for those exceptions for the time-being

Can't do this, I have to assume these are needed when using just uBO alone, not everybody out there has uBO-Extra installed, I suspect not many even know of its existence.

**Comment by gorhill** [comment]
Try 2.16b0, if all is well I will push it to the store today. Despite the exception filters, at least the browser is not abused with uBO-Extra. Without it the CPU and bandwidth when visiting one of the IL-based sites are being seriously abused.

**Comment by X01X012013** [comment]
> Can't do this, I have to assume these are needed when using just uBO alone

@gorhill As far as I tested, those rules are pretty useless.

**Comment by uBlock-user** [comment]
@gorhill Would you add `legacy.com` to the patch ? It's also fetching **g00** on specific pages. For example - www.legacy.com/obituaries/ottawacitizen/obituary.aspx?n=margaret-mollot&pid=144964521

**Comment by gorhill** [comment]
I just spotted a mistake in my regex which is used to remove the i10c code. That the fix worked with this mistake is still a mystery to me. I will be pushing 2.19, and it would be nice to have help to test the fix against all known i10c sites.

**Comment by uBlock-user** [comment]
g00 is back. URL - sandiegouniontribune.com uBO-Extra - `2.19` Chromium 59

Now it sends a bit of g00 and ends the loading of the page, so thumbnails of videos and pics won't load.

**Comment by X01X012013** [comment]
`I10C` is now renamed to `IXC`...

**Comment by okiehsch** [comment]
The funny thing is with Chrome 57 I still get no redirect or `g00` stuff at all with uBO-Extra enabled.
It even works for sites not listed in uBO-Extra.
For example `ottawasun.com` uses Instart Logic, without uBO-Extra I get the redirection, the cookies etc.
If I enable uBO-Extra none of that, even though `ottawasun.com` is not part of uBO-Extra.

**Comment by uBlock-user** [comment]
They're imitating the behavior of blocking `inline-script`.

**Comment by okiehsch** [comment]
@uBlock-user that is strange, they still do for me without uBlock-Extra

@gorhill 
> test the fix against all known i10c sites.

I did test all sites listed under Instart Logic v1. with `uBO-Extra 2.19` and there is
no obvious site breakage on my end.

**Comment by ghajini** [comment]
@gorhill 
old problem is back, videos broken
chrome 57,ubo 2.19

**Comment by okiehsch** [comment]
@ghajini that isssue is fixed on your end with uBO-Extra 2.18?
I ask because I can reproduce this issue with uBO-Extra 2.12+.

**Comment by gorhill** [comment]
> old problem is back, videos broken www.ottawacitizen.com
> chrome 57,ubo 2.19

I can reproduce the issue with Chromium 57. Interestingly, all works fine with Chrome 59.

**Comment by X01X012013** [comment]
@gorhill Can we just redefine user agent string to match what ever browser that is not affected? They probably have 10 different scripts and swapping them around to throw us off.

**Comment by gorhill** [comment]
If I add `ottawacitizen.com` to v1 buster without removing it from v2 buster, the video works.

There are so many exception filters for this site that have been added into EasyList...

**Comment by gorhill** [comment]
> Can we just redefine user agent string to match what ever browser that is not affected?

I worry this could interfere with legitimate uses of the user agent info in code other than the IL one.

**Comment by X01X012013** [comment]
@gorhill 
We don't have to apply the user agent patch to all IL websites, only ones that is not currently working. And I really don't think Chrome 57 to Chromium 59 would break anything else.

**Comment by PekaFat** [comment]
@gorhill can you fix wowhead.com? With add this `wowhead.com##script:inject(abort-on-property-write.js, MonkeyBroker)` to uBO filters, please?

With this script:
![image](https://cloud.githubusercontent.com/assets/13280748/25061360/6813a6b8-21c5-11e7-9dfe-dfcefa873ebc.png)


Without:
![image](https://cloud.githubusercontent.com/assets/13280748/25061362/806a313c-21c5-11e7-94d7-88690a067f93.png)


Google Chrome 57.0.2987.133
uBO 1.12.0
uBO-Extra 2.21
Filters:
![image](https://cloud.githubusercontent.com/assets/13280748/25061369/aded4180-21c5-11e7-8a43-bf5c4007bfe3.png)

**Comment by gorhill** [comment]
I was trying to find the original issue, and for whatever reason the issue has been removed from GitHub:
https://github.com/gorhill/uBO-Extra/issues/37

I asked GitHub why is this, waiting for an answer.

**Comment by PekaFat** [comment]
You can't add this rule without this issue?

**Comment by gorhill** [comment]
This is the answer I received from GitHub:

> When users are flagged by our system, their contributions are removed from view. The user who opened this issue can get more information about this situation by reaching out to us directly.

So someone flagged whoever opened <https://github.com/gorhill/uBO-Extra/issues/37>.

I added the filter, I just wanted to see the past conversation about it to see if there was any issue with it, or if something else worked.

**Comment by gotitbro** [comment]
https://www.lifewire.com is also inserting ads using g00.

**SS:** https://i.imgur.com/Cn6lUTm.png
**Error URL:** https://www.lifewire.com/wikipedia-3482753

**Comment by gotitbro** [comment]
@gorhill Please look into the above issue, thank you.

**Comment by gorhill** [comment]
Does this work?

    ||lifewire.com^$inline-script

**Comment by uBlock-user** [comment]
CNET is now sending /g00 - https://www.cnet.com/news/samsung-galaxy-note-8-launch-reportedly-slated-for-august/

**Comment by jspenguin2017** [comment]
@uBlock-user I can't quite reproduce that, and it doesn't refresh when I open DevTools.

**Comment by uBlock-user** [comment]
@jspenguin2017 Yeah well you may need to try multiple times to get the IL-crap loaded and executed first

**Comment by jspenguin2017** [comment]
Race condition? Where is the solution? If it's in uBO-Extra then there is no race condition.

**Comment by uBlock-user** [comment]
The domain's NOT added in uBO-Extra, that's why it still manages to execute the code and the code is the at the very top.

**Comment by jspenguin2017** [comment]
@uBlock-user I mean where is the solution? What is attempting to block the code to run?

**Comment by uBlock-user** [comment]
uBO-Extra and sometimes the script itself. It acts weirdly on some websites which are not added yet sometimes you cannot reproduce it while on some you can, not to mention Chromium version is also a contributing factor. Like for instance IL v2 doesn't work on my setup even when I disable uBO-Extra. I'm only affected by IL v1 websites from the added list.

**Comment by jspenguin2017** [comment]
@uBlock-user uBO-Extra will definitely not do anything if the website is not in the match list. (Except WebRTC anti-abuser, that is generic). 
Also, I pasted uBO-Extra to the end of my extension, I'm not sure what will happen if you enable both...

**Comment by uBlock-user** [comment]
Well then it's the script.

**Comment by jspenguin2017** [comment]
Maybe we should just edit the user agent to say something like Chrome 70 so the script just aborts?

**Comment by uBlock-user** [comment]
You're implying that by incrementing Chromium version to a non-existing number would make it abort ? Does that work in principle ?

**Comment by jspenguin2017** [comment]
The code tries to avoid analysis, Chromium may add an option to disable fetching details when the DevTools is open, so it needs to know which version of Chromium is "safe", it obviously won't work in general.

**Comment by jjohns71** [comment]
https://realclearpolitics.com seems to be using g00 adware now. It doesn't seem to be on their other 'realclear' sites for now, but I wouldn't be surprised if it migrates to them at some point in the future if I were to guess.

Screenshot with a sample of the logger is below:

![g00](https://user-images.githubusercontent.com/6795795/27620257-58de94f4-5b96-11e7-9e61-7fcd12114d45.PNG)

**Comment by ghajini** [comment]
new site using g00
`http://www.wrestlezone.com/news`
@gorhill

**Comment by simo1994** [comment]
I'm just leaving this here for future reference **IF** they manage to break even uBOExtra.
There is an official Google extension that permits users to spoof their user agent completely, overwriting the navigator.userAgent variable too.
https://chrome.google.com/webstore/detail/user-agent-switcher-for-c/djflhoibgkdhkhhcedjiklpkjnoahfmg

The usage is pretty straightforward. You insert a custom user-agent string, e.g. Firefox's, than you add on the "Permanent Spoof List" the domain `.` (just one dot, it will work on every domain) or you could add specific domains you know are using the g00 thing. Unfortunately, this has to be done manually.

However, you will probably miss out on some site improvements which are designed for Chrome and you might be logged out from some sites.  Don't be creative with user agent strings as you may be blacklisted, please use an actual third-party browser UA string.

**Comment by ghajini** [comment]
element picker doesn't seem to work
steps to produce=
visit `www.sandiegotribune.com`
click on news
enter element picker mode,element picker not opened


video=
https://streamable.com/6bajw

**Comment by jspenguin2017** [comment]
@ghajini That happens when the HTML is replaced by uBO-Extra.

**Comment by gotitbro** [comment]
@jspenguin2017 Can you could please elaborate on your above comment?

**Comment by jspenguin2017** [comment]
@gotitbro When uBO-Extra aborts the page to replace it, it has the side effect to break element picker. I'm not exactly sure why, I'm guessing that uBO will only inject the element picker once, and it gets overwritten.

**Comment by ghajini** [comment]
`tv.com`
 hosting g00 .......make ublock extra ***a rolling release*** same as ublock protector

**Comment by Hrxn** [comment]
Okay, so this g00 crap is to blame for extremely slowing down some sites, apparently.
I've been busy with restoring a couple of old computers, and it is ridiculous how drastic the difference is on some older system. With new hardware it is probably barely noticeable..

What is the best solution right now? Is there a specific filter? Or using uBO-Extra?
Or uBlock Protector? What would you recommend?

**Comment by jspenguin2017** [comment]
For old computers, Firefox is probably better. Firefox is hard on CPU and Chrome is hard on RAM, if you don't have enough RAM, then Firefox will be faster. g00 currently bails out on Firefox, not sure how long will that last though.

**Comment by Hrxn** [comment]
Based on my tests, Chrome is definitely better in this scenario. RAM is less of an issue, it's actually the CPU you will notice the most. Unless you have really ridiculously low RAM.

Just on example, really big single HTML documents are a problem, Firefox starts choking very easily while Chrome is still running smoothly.
Just opening tabs to quite heavy sites (news sites, nowadays) or sites with lots of embedded GIFs slow down Chrome noticeably.

Anyway, I was more interested in how to counter that g00 crap, actually 😄

**Comment by okiehsch** [comment]
The best way to counter it is to use Firefox, or if you prefer Chrome use uBO-Extra.

**Comment by jspenguin2017** [comment]
@okiehsch I wouldn't say so, the script can totally affect Firefox, it is just intentionally aborting on Firefox. It is a switch flip for it to start affecting Firefox.

**Comment by okiehsch** [comment]
I agree, I meant it is the best way to deal with IL-sites **right now**.

**Comment by gorhill** [comment]
> the script can totally affect Firefox

The IL scripts are not served with Firefox, just compare `view-source:http://www.tv.com/` with Firefox vs Chrome.

**Comment by jspenguin2017** [comment]
> The IL scripts are not served with Firefox

You are saying they cannot serve it to Firefox? I highly doubt that. An user agent will accept anything the server sends.

**Comment by gorhill** [comment]
> You are saying they cannot serve it to Firefox?

Nowhere did I say this.

**Comment by Hrxn** [comment]
Okay, to conclude, Firefox has the benefit of being unaffected by this plague, at least for now.

And with Chrome (which works better on the old systems here), it is uBlock Origin + uBO-Extra, in preference to just uBO or uBO + uBlock Protector.
Agreed?

**Comment by jspenguin2017** [comment]
uBlock Protector defuses anti-adblock and has uBO-Extra embedded in it. If websites you use don't have anti-adblock then uBlock Origin + uBO-Extra will be faster.

**Comment by uBlock-user** [comment]
`g00` infested websites can now be blocked via `abort-current-inline-script.js`

Example - `sandiegouniontribune.com##script:inject(abort-current-inline-script.js, atob, g00)`

**Comment by Hrxn** [comment]
Does not seem to work for metacritic.com, for example.
`metacritic.com##script:inject(abort-current-inline-script.js, atob, g00)`

First load did work, I think, but doing a hard reload shows old behaviour.

**Comment by uBlock-user** [comment]
I'm not receiving any `g00` on `metacritic.com` even after disabling uBO-Extra, I'm on Chromium 62.

**Comment by uBlock-user** [comment]
Tested on `orlandosentinel.com` and `chicagotribune.com`. Working as expected.

**Comment by uBlock-user** [comment]
Tested on `mcall.com`, `boston.com` and `sun-sentinal.com`. Working as expected.

**Comment by Hrxn** [comment]
Chrome 60.0.3112.113 (64-bit) (Stable, latest version), uBO 1.13.8, _without_ uBO-Extra.

> `g00` can now be blocked via `abort-current-inline-script.js`

I assumed this meant uBO-Extra would not be required here, sorry if I misunderstood this.

**Comment by uBlock-user** [comment]
Yes, that's exactly what I meant, however you may need uBO-Extra for blocking WebRTC connections.

**Comment by gorhill** [comment]
> I haven't tested all but feel free.

It does not work, that's the entire point of uBO-Extra. That kind of inaccurate information just add noise and confuse further anybody reading this now too huge thread. I already explained in the past why uBO-Extra is needed.

Locking this thread. For any issue with IL stuff, open a new issue.

Stick to observed facts please, otherwise this just contribute to propagate myths out there.

**Comment by gorhill** [comment]
@jspenguin2017 Sorry, you are correct about IL stuff being served on Firefox, I just tested many sites, and I see the IL script at the top with Firefox. I believe this is a new development, I am pretty sure this was not the case until no long ago. So possibly uBO-Extra will be needed for Firefox in the near future.

In any case, I rather discuss all this at uBO-Extra itself, the issue has grown too large and anyways it was solved long ago with the release of uBO-Extra.
